### PR TITLE
Added more detail to spiffe-tls example readme

### PR DESCRIPTION
**Pull request by martincapello**

### CODE REVIEW

File: v2/examples/spiffe-tls/README.md [hunk 1]
```diff
@@ -1,73 +1,81 @@
...
+This example shows how to use the go-spiffe library to make two workloads establish an mTLS connection using X.509 SVIDs obtained from the SPIFFE Workload API. 
 
-One of the workloads acts as a client and the other as the server. The use the SPIRE default socket path: `/tmp/agent.sock`. This value can also be set via the `SPIFFE_ENDOINT_SOCKET` environment variable.
+One workload acts as a client and the other as the server. 
 
-The **server workload** creates a listener using the [spiffetls.Listen](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/spiffetls?tab=doc#Listen) function.
-It uses [spiffetls.MTLSServerWithSourceOptions](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/spiffetls?tab=doc#MTLSServerWithSourceOptions) to configure  the source, and validates that clients present an X509-SVID with an expected SPIFFE ID.
+The scenario goes like this:
+1. The server starts listening for incoming SPIFFE-compliant mTLS connections.
+2. The client establish an SPIFFE-compliant mTLS connection to the server.
```
Comment (sanderson042) [inline v2/examples/spiffe-tls/README.md:9]: establish -> establishes

File: v2/examples/spiffe-tls/README.md [hunk 2]
```diff
@@ -1,73 +1,81 @@
...
 
-One of the workloads acts as a client and the other as the server. The use the SPIRE default socket path: `/tmp/agent.sock`. This value can also be set via the `SPIFFE_ENDOINT_SOCKET` environment variable.
+One workload acts as a client and the other as the server. 
 
-The **server workload** creates a listener using the [spiffetls.Listen](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/spiffetls?tab=doc#Listen) function.
-It uses [spiffetls.MTLSServerWithSourceOptions](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/spiffetls?tab=doc#MTLSServerWithSourceOptions) to configure  the source, and validates that clients present an X509-SVID with an expected SPIFFE ID.
+The scenario goes like this:
+1. The server starts listening for incoming SPIFFE-compliant mTLS connections.
+2. The client establish an SPIFFE-compliant mTLS connection to the server. 
+3. The server starts waiting a message from the client.
```
Comment (sanderson042) [inline v2/examples/spiffe-tls/README.md:10]: waiting a -> waiting for a

File: v2/examples/spiffe-tls/README.md [hunk 3]
```diff
@@ -1,73 +1,81 @@
...
+4. The client sends a "Hello server" message and starts waiting a response.
```
Comment (sanderson042) [inline v2/examples/spiffe-tls/README.md:11]: waiting a -> waiting for a

File: v2/examples/spiffe-tls/README.md [hunk 4]
```diff
@@ -1,73 +1,81 @@
...
-
+Where:
+- ctx is a `context.Context`. `ListenWithMode` blocks until the first Workload API response is received or this context times out or is cancelled.
+- serverAddress is the address (`localhost:55555`) where the server workload is going to listen for client connections.
+- [spiffetls.MTLSServerWithSourceOptions](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/spiffetls?tab=doc#MTLSServerWithSourceOptions) is used to configure the [X509Source](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2@v2.0.0-alpha.3/workloadapi?tab=doc#X509Source) used by the internal Workload API client.
+- clientID is a SPIFFE ID (`spiffe://example.org/client`), which along with the [tlsconfig.AuthorizeID](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig?tab=doc#AuthorizeID) function makes the server to accept only clients that present an X509-SVID having such identity. You can pick any of the functions that return a [tlsconfig.Authorizer](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig?tab=doc#Authorizer) included with the library, or you can make your own.
```
Comment (sanderson042) [inline v2/examples/spiffe-tls/README.md:28]: function makes the server to accept only clients that present an X509-SVID having such identity -> function configures the server to accept only clients that present an X509-SVID with a matching SPIFFE ID

Here I'm not sure if I understood this correctly, so my suggestion may be wrong.

File: v2/examples/spiffe-tls/README.md [hunk 5]
```diff
@@ -1,73 +1,81 @@
...
-
+Where:
+- ctx is a `context.Context`. `DialWithMode` blocks until the first Workload API response is received or this context times out or is cancelled.
+- serverAddress is the address (`localhost:55555`) where the server workload is listening for client connections.
+- [spiffetls.MTLSClientWithSourceOptions](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/spiffetls?tab=doc#MTLSClientWithSourceOptions) is used to configure the [X509Source](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2@v2.0.0-alpha.3/workloadapi?tab=doc#X509Source) used by the internal Workload API client.
+- spiffeID is a SPIFFE ID (`spiffe://example.org/server`), which along with the [tlsconfig.AuthorizeID](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig?tab=doc#AuthorizeID) function makes the client to connect only to a server that presents an X509-SVID having such identity. You can pick any of the functions that return a [tlsconfig.Authorizer](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig?tab=doc#Authorizer) included with the library, or you can make your own.
```
Comment (sanderson042) [inline v2/examples/spiffe-tls/README.md:47]: Same suggestion as the last one, if it makes sense.

File: v2/examples/spiffe-tls/README.md [hunk 6]
```diff
@@ -1,73 +1,81 @@
...
-The [tlsconfig.Authorizer](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig?tab=doc#Authorizer) is used to authorize the mTLS peer. In this example, both the client and server use it to authorize the specific SPIFFE ID of the other side of the connection.
-
-That is it! The go-spiffe library fetches and automatically renews the X.509 SVIDs of both workloads from the Workload API provider (i.e. SPIRE).
-
-As soon as the mTLS connection is established, the client sends a message to the server and gets a response.
+	conn, err := spiffetls.Dial(ctx, "tcp", serverAddress, tlsconfig.AuthorizeID(spiffeID))
+```
 
+## That is it!
+As we can see the go-spiffe library allows your application to use the Workload API transparently for both ends of the connection, it takes care of fetching and automatically renewing the X.509 SVIDs needed to mantain a secure communication.
```
Comment (sanderson042) [inline v2/examples/spiffe-tls/README.md:54]: connection, it -> connection. The go-spiffe library

File: v2/examples/spiffe-tls/README.md [hunk 7]
```diff
@@ -1,73 +1,81 @@
...
 ## Running
 This example assumes the following preconditions:
-- There is a SPIRE server and agent up and running.
+- There is a SPIRE server and a SPIRE agent up and running.
 - There is a Unix workload attestor configured.
-- The trust domain is `example.org`
-- The agent SPIFFE ID is `spiffe://example.org/host`.
-- There is a `server-workload` and `client-workload` user in the system.
+- The trust domain is `example.org`.
+- The agent'S SPIFFE ID is `spiffe://example.org/host`.
```
Comment (sanderson042) [inline v2/examples/spiffe-tls/README.md:74]: 'S -> 's

(case)

File: v2/examples/spiffe-tls/README.md [hunk 8]
```diff
@@ -97,15 +105,15 @@ sudo -u client-workload ./client
 
 The server should have got a _"Hello server"_ message and responded with a _"Hello client"_ message.
```
Comment (sanderson042) [inline v2/examples/spiffe-tls/README.md:106]: got -> received

File: v2/examples/spiffe-tls/README.md [hunk 9]
```diff
@@ -97,15 +105,15 @@ sudo -u client-workload ./client
...
+If either workload encounters a peer with a different SPIFFE ID than the one it expects, they abort the TLS handshake and the connection fails.  
+For instance, when running the client with the server's user: 
 ```
 sudo -u server-workload ./client
 
 Unable to read server response: remote error: tls: bad certificate
 ```
 
-And the server log shows:
+And the server logs:
```
Comment (sanderson042) [inline v2/examples/spiffe-tls/README.md:116]: \* -> The server log would contain:

File: v2/examples/spiffe-tls/README.md [hunk 10]
```diff
@@ -97,15 +105,15 @@ sudo -u client-workload ./client
 
 The server should have got a _"Hello server"_ message and responded with a _"Hello client"_ message.
 
-If either workload encounters a peer with a different SPIFFE ID, they will abort the TLS handshake and the connection will fail. 
- 
+If either workload encounters a peer with a different SPIFFE ID than the one it expects, they abort the TLS handshake and the connection fails.
```
Comment (sanderson042) [inline v2/examples/spiffe-tls/README.md:108]: they abort -> the workload aborts

File: v2/examples/spiffe-tls/README.md (hunk 3 above)
Comment (martincapello) [reply v2/examples/spiffe-tls/README.md:11]: At leat I wrote it consistently wrong :sweat_smile:

File: v2/examples/spiffe-tls/README.md (hunk 4 above)
Comment (martincapello) [reply v2/examples/spiffe-tls/README.md:28]: You understood it correctly, your suggestion is just perfect, thanks!

**Comment by martincapello** [comment]
Great suggestions/fixes! I want to take the opportunity to let you know that https://github.com/spiffe/spiffe.io/pull/144 is somehow a related change.

File: v2/examples/spiffe-tls/README.md [hunk 11]
```diff
@@ -1,73 +1,81 @@
 # Mutually Authenticated TLS (mTLS)
 
-This example shows how two services can communicate using mTLS with X.509 SVIDs obtained from the SPIFFE Workload API. 
+This example shows how to use the go-spiffe library to make two workloads establish an mTLS connection using X.509 SVIDs obtained from the SPIFFE Workload API.
```
Comment (azdagron) [inline v2/examples/spiffe-tls/README.md:3]:
```suggestion
This example shows how to use the go-spiffe library to establish an mTLS connection between two workloads using X.509 SVIDs obtained from the SPIFFE Workload API. 
```

File: v2/examples/spiffe-tls/README.md [hunk 12]
```diff
@@ -1,73 +1,81 @@
...
+- clientID is a SPIFFE ID (`spiffe://example.org/client`), which along with the [tlsconfig.AuthorizeID](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig?tab=doc#AuthorizeID) function configures the server to accept only clients that present an X509-SVID with a matching SPIFFE ID. You can pick any of the functions that return a [tlsconfig.Authorizer](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig?tab=doc#Authorizer) included with the library, or you can make your own. 
+- socketPath is the address of the Workload API (`unix:///tmp/agent.sock`) to which the internal Workload API client connects to get up-to-date SVIDs. Alternatively, we could have omitted this configuration option, in which case the listener would have used the `SPIFFE_ENDPOINT_SOCKET` environment variable to locate the Workload API, then the code could have been written like this:
```
Comment (azdagron) [inline v2/examples/spiffe-tls/README.md:29]:
```suggestion
- socketPath is the address of the Workload API (`unix:///tmp/agent.sock`) to which the internal Workload API client connects to get up-to-date SVIDs. Alternatively, we could have omitted this configuration option, in which case the listener would have used the `SPIFFE_ENDPOINT_SOCKET` environment variable to locate the Workload API. The code could have then been written like this:
```

File: v2/examples/spiffe-tls/README.md [hunk 13]
```diff
@@ -1,73 +1,81 @@
...
+- spiffeID is a SPIFFE ID (`spiffe://example.org/server`), which along with the [tlsconfig.AuthorizeID](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig?tab=doc#AuthorizeID) function configures the client to connect only to a server that presents an X509-SVID with a matching SPIFFE ID. You can pick any of the functions that return a [tlsconfig.Authorizer](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig?tab=doc#Authorizer) included with the library, or you can make your own. 
+- socketPath is the address of the Workload API (`unix:///tmp/agent.sock`) to which the internal Workload API client connects to get up-to-date SVIDs. Alternatively, we could have omitted this configuration option, in which case the dialer would have used the `SPIFFE_ENDPOINT_SOCKET` environment variable to locate the Workload API, then the code could have been written like this:
```
Comment (azdagron) [inline v2/examples/spiffe-tls/README.md:48]:
```suggestion
- socketPath is the address of the Workload API (`unix:///tmp/agent.sock`) to which the internal Workload API client connects to get up-to-date SVIDs. Alternatively, we could have omitted this configuration option, in which case the dialer would have used the `SPIFFE_ENDPOINT_SOCKET` environment variable to locate the Workload API, The code could have then been written like this:
```

File: v2/examples/spiffe-tls/README.md [hunk 14]
```diff
@@ -1,73 +1,81 @@
...
-The [tlsconfig.Authorizer](https://pkg.go.dev/github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig?tab=doc#Authorizer) is used to authorize the mTLS peer. In this example, both the client and server use it to authorize the specific SPIFFE ID of the other side of the connection.
-
-That is it! The go-spiffe library fetches and automatically renews the X.509 SVIDs of both workloads from the Workload API provider (i.e. SPIRE).
-
-As soon as the mTLS connection is established, the client sends a message to the server and gets a response.
+	conn, err := spiffetls.Dial(ctx, "tcp", serverAddress, tlsconfig.AuthorizeID(spiffeID))
+```
 
+## That is it!
+As we can see the go-spiffe library allows your application to use the Workload API transparently for both ends of the connection. The go-spiffe takes care of fetching and automatically renewing the X.509 SVIDs needed to mantain a secure communication.
```
Comment (azdagron) [inline v2/examples/spiffe-tls/README.md:54]:
```suggestion
As we can see the go-spiffe library allows your application to use the Workload API transparently for both ends of the connection. The go-spiffe library takes care of fetching and automatically renewing the X.509 SVIDs needed to maintain a secure communication.
```

Comment (martincapello) [reply v2/examples/spiffe-tls/README.md:54]: :dart:

**Pull request merged**